test/e2e-tests-ingress.sh

```diff
@@ -0,0 +1,115 @@
+  echo -n "Waiting until taskrun $1 completed\n"
+  for i in {1..150}; do  # timeout after 5 minutes
+    taskrun_status="$(kubectl get taskrun $1 -o=jsonpath='{.status.conditions[0]}')"
+    match=$(echo $taskrun_status | grep "message:All Steps have completed executing reason:Succeeded status:True type:Succeeded")  || true
+    l=${#match}
+    if [ 0 -ne $l ];
+    then
+      return 0
+    fi
+    sleep 2
+  done
+  echo -e "\n\nERROR: timeout waiting for taskrun successful completion\n"
+  return 1
+}
+
+# Waits until pod started.
+# Parameters: $1 - pod name prefix
+function wait_until_pod_started() {
+  echo -n "Waiting until pod started\n"
+  for i in {1..150}; do  # timeout after 5 minutes
+    pod_status=$(kubectl get pod | grep $1 | grep "Running") || true
+    l=${#pod_status}
+    if [ 0 -ne $l ];
+    then
+      return 0
+    fi
+    sleep 2
+  done
+  echo -e "\n\nERROR: timeout waiting for pod successful start\n"
+  return 1
+}
+
+set -o errexit
+set -o pipefail
+
+# verify if the yaml file is valid
+for op in apply delete;do 
+  kubectl ${op} -f ${REPO_ROOT_DIR}/docs/create-webhook.yaml
+done
+
+# make sure no remaining resources from the previous run
+echo "clean up before start. Ignore (NotFound) errors"
+kubectl delete secret secret1 || true
+kubectl delete eventlistener listener || true
+kubectl delete taskrun create-webhook || true
+kubectl delete secret githubsecret || true
+
+# setup
+kubectl apply -f ${REPO_ROOT_DIR}/test/ingress-clusterrole.yaml
+kubectl apply -f ${REPO_ROOT_DIR}/test/ingress-clusterrolebinding.yaml
+kubectl apply -f ${REPO_ROOT_DIR}/docs/create-webhook.yaml
+kubectl create secret generic githubsecret --from-literal=accessToken=ff7d2c2949844f68cb18a68f4febad4454df2336 --from-literal=userName=tektonuser
+
+
+# test
+kubectl apply -f ${REPO_ROOT_DIR}/docs/create-webhook-run.yaml
+wait_until_taskrun_completed create-webhook 
+
+# check certificate
+crt=$(kubectl get secret secret1 -o=jsonpath='{.data.tls\.crt}')
+echo $crt | base64 --decode | grep "\-\-\-\-\-BEGIN CERTIFICATE\-\-\-\-\-"
+echo $crt | base64 --decode | grep "\-\-\-\-\-END CERTIFICATE\-\-\-\-\-"
+
+key=$(kubectl get secret secret1 -o=jsonpath='{.data.tls\.key}')
+echo $key | base64 --decode | grep "\-\-\-\-\-BEGIN RSA PRIVATE KEY\-\-\-\-\-"
+echo $key | base64 --decode | grep "\-\-\-\-\-END RSA PRIVATE KEY\-\-\-\-\-"
+
+# check ingress
+svc=$(kubectl get ingress listener -o=jsonpath='{.spec.rules[0].http.paths[0].backend.serviceName}')
+host=$(kubectl get ingress listener -o=jsonpath='{.spec.rules[0].host}')
+tlshost=$(kubectl get ingress listener -o=jsonpath='{.spec.tls[0].hosts[0]}')
+secret=$(kubectl get ingress listener -o=jsonpath='{.spec.tls[0].secretName}')
+if [ $svc != "listener" ] || [ $host != "listener.192.168.0.1.nip.io" ] || [ $tlshost != "listener.192.168.0.1.nip.io" ] || [ $secret != "secret1" ]; then
+  echo -e "unexpected values " $svc $host $tlshost $secret
+  exit 1
+fi
+
+# check webhook
```

Comment:
nit: Please remove this comment since we aren't testing the webhook creation part right now.

Reply:
yes.